plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -716,6 +721,10 @@ public static int parseVersion(String metadataFileName)
         if (matcher.matches()) {
             return parseInt(matcher.group("version"));
         }
+        matcher = HADOOP_GENERATED_METADATA_FILE_NAME_PATTERN.matcher(metadataFileName);
```

Comment:
This change will allow having a mixture of files:

- `0000x-uuid.metadata.json`
- `vx.metadata.json` 

As the test shows is this is actually possible when registering a Hadoop table.
I think it is a legal to assume that both kind of file name formats may exist in the `metadata/` directory.

:+1: